spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestSnapshotSelection.java

```diff
@@ -425,16 +426,35 @@ public void testSnapshotSelectionByBranchWithSchemaChange() throws IOException {
     // Deleting a column to indicate schema change
     table.updateSchema().deleteColumn("data").commit();
 
-    // The data should have the deleted column as it was captured in an earlier snapshot.
-    Dataset<Row> deletedColumnBranchSnapshotResult =
-        spark.read().format("iceberg").option("branch", "branch").load(tableLocation);
-    List<SimpleRecord> deletedColumnBranchSnapshotRecords =
-        deletedColumnBranchSnapshotResult
-            .orderBy("id")
-            .as(Encoders.bean(SimpleRecord.class))
-            .collectAsList();
-    Assert.assertEquals(
-        "Current snapshot rows should match", expectedRecords, deletedColumnBranchSnapshotRecords);
+    // The data should not have the deleted column
+    Assertions.assertThat(
```

Comment:
I think that we also need a write test to ensure that we can write to the branch with the current table schema.

Reply:
good point, I've done that